Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -156,32 +150,30 @@ - (instancetype)initWithStorageManager:(TSStorageManager *)storageManager
     OWSAssert([NSThread isMainThread]);
     OWSAssert(storageManager);
     OWSAssert(messageSender);
+    OWSAssert(messageSender);
 
     _messageSender = messageSender;
     _dbConnection = storageManager.newDatabaseConnection;
+    _networkManager = networkManager;
+
     _userProfileWhitelistCache = [NSMutableDictionary new];
     _groupProfileWhitelistCache = [NSMutableDictionary new];
-    _otherUsersProfileKeyCache = [NSMutableDictionary new];
     _otherUsersProfileAvatarImageCache = [NSCache new];
 
     OWSSingletonAssert();
 
-    // Register this manager with the message sender.
-    // This is a circular dependency.
-    [messageSender setProfilesManager:self];
-
-    // Try to load.
-    _localProfileKey = [self.dbConnection objectForKey:kOWSProfilesManager_LocalProfileSecretKey
-                                            inCollection:kOWSProfilesManager_Collection];
-    if (!_localProfileKey) {
-        // Generate
-        _localProfileKey = [OWSProfilesManager generateLocalProfileKey];
-        // Persist
-        [self.dbConnection setObject:_localProfileKey
-                                forKey:kOWSProfilesManager_LocalProfileSecretKey
-                          inCollection:kOWSProfilesManager_Collection];
+    self.localUserProfile = [self getOrCreateUserProfileForRecipientId:kNSNotificationName_LocalProfileUniqueId];
```

Comment:
Wait... this should supercede the old way that you're referring to, where the key was the actual local phone number.  I wanted to make sure that we could, for example, safely generate this key _before_ the user registered, in the interest of decoupling.

On the other hand, I haven't special-cased "self" (at least not yet) so if we have a conversation with ourself, we'll load our profile.  It wouldn't be too hard to special-case if you want to do that.